### ISSUE TITLE
tag names containing hyphens prevent wild card searches on measurements containing them

### ISSUE THREAD

**Issue by jonseymour** [influxdb/influxdb#3683]
Test Case
========

```drop measurement cpu```

write the following data into the cpu measurement:

```cpu,host=A,region=US value=124 1234501```

confirm that the following works:

```select * from cpu```

write the following data into the cpu measurement:

```cpu,host=A,the-region=US value=124 1234501```

confirm that the following:

```select * from cpu```

yields:

```Server returned error: only time and tag dimensions allowed```

Expected result
============

Either:

- /write fails because tag names contain unsupported characters such as hyphen
- select is changed to allow tag names containing these characters

**Comment by jonseymour** [comment]
Another variant of this problem is a tag name which is literally "user".

So:

```cpu,host=A,region=US,userx=fred value=124 1234501```

works as expected, but:

```cpu,host=A,region=US,user=fred value=124 1234501```

does not, resulting in:

```Server returned error: found USER, expected identifier, string, number, bool at line 1, char 52```

**Comment by jonseymour** [comment]
Another variant of this problem exists if the field name contains hyphens or is literally 'user'

**Comment by yvesf** [comment]
I tried with build from git HEAD and wasn't able to reproduce:
```
curl 'http://localhost:8086/write?db=test' --data-binary 'cpu,host=A,region=US,user=fred value=1234501 1234501'
curl 'http://localhost:8086/write?db=test' --data-binary 'cpu,host=A,region=US,userx=fred value=2345012 2345012'
curl 'http://localhost:8086/write?db=test' --data-binary 'cpu,host=A,region=US,user-x=fred value=3450123 3450123'
echo -e "use test\n SELECT * FROM cpu" | influx
```
```
name: cpu
tags: host=A, region=US, user-x=, user=fred, userx=
time				value
----				-----
1970-01-01T00:00:00.001234501Z	1.234501e+06
name: cpu
tags: host=A, region=US, user-x=fred, user=, userx=
time				value
----				-----
1970-01-01T00:00:00.003450123Z	3.450123e+06
name: cpu
tags: host=A, region=US, user-x=, user=, userx=fred
time				value
----				-----
1970-01-01T00:00:00.002345012Z	2.345012e+06
```

Did I made a mistake or might it be that it's fixed in the meantime?

**Comment by jonseymour** [comment]
Just updated to c2dbf16 and I can't reproduce either. It has been fixed since a09c00653
